api/defaultHttpClient.go

```diff
@@ -0,0 +1,30 @@
+package api
+
+import (
+	"net/http"
+	"sync"
+	"time"
+)
+
+type httpClient struct {
+	client *http.Client
+}
+
+// NewDefaultHttpClient will create a default http client which can be queried to GET url responses
+func NewDefaultHttpClient(requestTimeoutSec uint64) *httpClient {
```

Comment:
i think we can make some checks on `requestTimeoutSec` to be greater than 0 or something

Reply:
Actually, zero time out is valid, as specified in `http` library. Added extra comment in `config.toml` to clarify it